yt_dlp/extractor/mxplayer.py

```diff
@@ -1,178 +1,99 @@
-            'episode': 'Episode 0'
-        },
-        'params': {
-            'format': 'best',
-            'skip_download': True,
-        },
     }]
 
     def _real_extract(self, url):
-        type, display_id, video_id = self._match_valid_url(url).groups()
-        type = 'movie_film' if type == 'movie' else 'tvshow_episode'
-        API_URL = 'https://androidapi.mxplay.com/v1/detail/'
-        headers = {
-            'X-Av-Code': '23',
-            'X-Country': 'IN',
-            'X-Platform': 'android',
-            'X-App-Version': '1370001318',
-            'X-Resolution': '3840x2160',
-        }
-        data_json = self._download_json(f'{API_URL}{type}/{video_id}', display_id, headers=headers)['profile']
-
-        season, series = None, None
-        for dct in data_json.get('levelInfos', []):
-            if dct.get('type') == 'tvshow_season':
-                season = dct.get('name')
-            elif dct.get('type') == 'tvshow_show':
-                series = dct.get('name')
-        thumbnails = []
-        for thumb in data_json.get('poster', []):
-            thumbnails.append({
-                'url': thumb.get('url'),
-                'width': thumb.get('width'),
-                'height': thumb.get('height'),
-            })
+        video_type, display_id, video_id = self._match_valid_url(url).groups()
+
+        if 'show' in video_type:
+            video_type = 'episode'
+
+        API_URL = f'https://api.mxplay.com/v1/web/detail/video?type={video_type}&id={video_id}'
+        STREAM_URL = 'https://llvod.mxplay.com/{}'
 
+        data_json = self._download_json(API_URL, display_id)
+
+        title = data_json['title']
+        duration = data_json.get('duration')
         formats = []
         subtitles = {}
-        for dct in data_json.get('playInfo', []):
-            if dct.get('extension') == 'mpd':
-                frmt, subs = self._extract_mpd_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
-            elif dct.get('extension') == 'm3u8':
-                frmt, subs = self._extract_m3u8_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
+        description = data_json.get('description')
+        series, season, season_number, episode_number = None, None, None, None
+
+        if video_type == 'episode':
+            series = data_json.get('container').get('container').get('title')
+            season = data_json.get('container').get('title')
+            episode_number = data_json.get('sequence')
+
+            if season:
+                match = re.match(r'Season (?P<season_number>\d+)', season)
+                if match:
+                    season_number = int(match.group('season_number'))
```

Comment:
I think you could use `self._search_regex()`

Reply:
Done.